libs/vm-modules/include/vm_modules/ml/model/model.hpp

```diff
@@ -118,6 +118,8 @@ class VMModel : public fetch::vm::Object
   ModelCategory      model_category_ = ModelCategory::NONE;
   bool               compiled_       = false;
 
+  static constexpr size_t min_allowed_hidden_layers_ = 2;
```

Comment:
this value should be 1; and you should fix the code that doesn't correctly handle that case

Reply:
also use std::size_t (code style)

Reply:
I would keep it 2 for now as it protects user from a runtime crash. Indeed we need to fix Graph code, but right now I can not estimate if that fix is small or huge (and what risk does it add), so I guess it might be better to create a separate task for this investigation, and set VMModel::min_allowed_hidden_layers_ to 1 in that scope.

Reply:
I leave a comment above this constant to explain the current value

Reply:
After clarification with @MissingNO57 : no changes are required to existing code, it works fine, just a naming confusion. I have renamed necessary constants and variables.